documentation/docs/examples/timeline.mdx

```diff
@@ -0,0 +1,213 @@
+    width: 0%;
+    height: 20px;
+    border-radius: 9px;
+    position: relative;
+  `;
+  return progress;
+}
+
+model.setDefaultNodeAppearance(DefaultNodeAppearance.Ghosted);
+
+const timeline = new TimelineTool(model);
+
+// div element Cognite3DViewer to insert into the canvas
+const domElement = document.getElementById('demo-wrapper');
+const container = createProgressBarContainer();
+const progressBar = createProgressBar();
+container.appendChild(progressBar);
+domElement.appendChild(container);
+
+const nodes = new AssetNodeCollection(sdk, model);
+const keyframeAssetNode = timeline.createKeyframe(new Date('2021-10-05'));
+keyframeAssetNode.assignStyledNodeCollection(nodes, { renderGhosted: false, outlineColor: NodeOutlineColor.Cyan });
+
+nodes.executeFilter({ assetId: 4222791488928479 });
+const keyframeAssetID = timeline.createKeyframe(new Date('2021-10-06'));
+keyframeAssetID.assignStyledNodeCollection(nodes, { renderGhosted: false, outlineColor: NodeOutlineColor.Orange });
+
+const allPipes = new PropertyFilterNodeCollection(sdk, model);
+allPipes.executeFilter({'PDMS': {'Type': 'PIPE'}});
+const keyframeAllPipes = timeline.createKeyframe(new Date('2021-10-07'));
+keyframeAllPipes.assignStyledNodeCollection(allPipes, { renderGhosted: false, color: [40, 200, 20] });
+
+const stuMechNodes = new PropertyFilterNodeCollection(sdk, model);
+stuMechNodes.executeFilter({'PDMS': {':capDisc': 'MECH', 'Purpose': 'STU' }});
+const keyframeMechNodes = timeline.createKeyframe(new Date('2021-10-08'));
+keyframeMechNodes.assignStyledNodeCollection(stuMechNodes, { renderGhosted: false, color: [80, 100, 120] });
+
+const playLength = 25000;
+
+timeline.play(new Date('2021-10-05'), new Date('2021-10-09'), playLength);
+
+let refDate = 0;
+
+timeline.subscribe('dateChanged', (data) => {
+  if(refDate === 0) {
+    refDate = data.date.getTime();
+  }
+  const width = ((data.date.getTime() - refDate) / playLength) * 100;
+  const widthPercentage = width.toString() + '%';
+  progressBar.style.width = widthPercentage;
+})
+```
+
+### Stop timeline playback
+
+```js runnable
+timeline.stop();
+```
+
+### Pause timeline
+
+```js runnable
+timeline.pause();
+```
+
+### Resume timeline
+
+```js runnable
+timeline.resume();
+```
+
+### Get Keyframe by Date
+`getKeyframeByDate(date: Date)`
+- `date` - Date of the Keyframe which needs retrived.
+
+```jsx
+const keyframe = timeline.getKeyframeByDate(new Date('2021-10-06'));
+```
+
+### Timeline Data's while in Play
+While in Play state, the tool emits few data which will be helpful for the users to calculate the progression of the Playback of Timeline.
+Tool provides `TimelineDateUpdateDelegate` event which can be subscribed to received below data.
+- `date` - Virtual Date which can be used to plot progression.
+- `activeKeyframe` - Current Keyframe in the Playback.
+- `startDate` - Keyframe date to start the Playback of Timeline Keyframes.
+- `endDate` - Keyframe date to stop the Playback of Timeline Keyframes.
```

Comment:
```suggestion
### Reacting to date changes while in playback mode
While in the play state, the tool emits events which will be helpful for the users to calculate the progression of the playback of the timeline.
The tool provides a `TimelineDateUpdateDelegate` event which can be subscribed to received below data.
- `date` - Virtual date which can be used to calculate progression.
- `activeKeyframe` - Current active `Keyframe` in the playback.
- `startDate` - Virtual date to start the playback of the timeline.
- `endDate` - Virtual date to stop the playback of the timeline.
```

Also, consider using these values in the example above (rather than storing startDate/endDate like I suggested there).

Note! There is the possibility of `activeKeyframe` not being set - e.g. if setting up the tool like this with keyframes on dates 2020-01-01, 2020-02-01, 2020-03-01 and then doing 
`play(2019-12-01, 2020-04-01,...)`. In this case there will be no active keyframes before 2020-01-01. I don't belive the tool accounts for this currently - correct me if I'm wrong